mypyc/irbuild/specialize.py

```diff
@@ -157,14 +157,19 @@ def translate_globals(builder: IRBuilder, expr: CallExpr, callee: RefExpr) -> Va
 
 
 @specialize_function("builtins.abs")
-def translate_abs(builder: IRBuilder, expr: CallExpr, callee: RefExpr) -> Value | None:
-    """Specialize calls on native classes that implement __abs__."""
-    if len(expr.args) == 1 and expr.arg_kinds == [ARG_POS]:
+@specialize_function("builtins.int")
+@specialize_function("builtins.float")
+def translate_builtins_with_unary_dunder(
+    builder: IRBuilder, expr: CallExpr, callee: RefExpr
+) -> Value | None:
+    """Specialize calls on native classes that implement the associated dunder."""
+    if len(expr.args) == 1 and expr.arg_kinds == [ARG_POS] and isinstance(callee, NameExpr):
         arg = expr.args[0]
         arg_typ = builder.node_type(arg)
-        if isinstance(arg_typ, RInstance) and arg_typ.class_ir.has_method("__abs__"):
+        method = f"__{callee.name}__"
```

Comment:
The implementation of `NameExpr` documents that `name` might be the qualified name. Is that going to be the case here?

Reply:
That comment is incorrect. `name` is always an unqualified name (no dots). Would you like to update the comment in this PR (optional)?

Reply:
Thank you for clarifying!

> Would you like to update the comment in this PR (optional)?

I'd love to. If it worried me, then it might worry someone else down the road, let's avoid that.